### DISCUSSION TITLE
Question about EDM or MEDM conversion, and relative folders.

### DISCUSSION THREAD

**Discussion by BoredDadDev** [ControlSystemStudio/phoebus#2741] [Q&A] [answered]
Hi Everyone, let me start by thanking you all for the immense effort put into developing Phoebus, and into helping less experienced users like me.

I just have a question regarding the conversion of EDM or MEDM files to .bob, and it's mostly related to file locations, and the folder structure.

At my workplace, say with EDM, we tend to just reference other screens by file name alone, from the main screen, and then add the paths where thoes other screens reside to the EDMDATAFILES environment variable before launching the main screen. This allows EDM to find the right files, even when inside a subfolder and referenced by just the name.
Example:
main_folder/main_screen.edl, references secondary_screen.edl, which is inside main_folder/secondary_folder/

This works as long as EDMDATAFILES contains the full path to the secondary_folder. In practice, there might be many such secondary folders.

My question is then, is there a way to tell Phoebus to use multiple folders when looking for the .bob files, in a similar manner?

I know, that when using auto-conversion, it's possible to specify all the secondary folders (say for EDM) and their conversion settings, as explained here,
https://control-system-studio.readthedocs.io/en/stable/app/display/convert-medm/doc/index.html?highlight=auto%20conversion#converting-edm-edl-displays

But that seems to solve a slightly different problem, where the files are referenced using relative paths in the main_screen.edl.

Note: In case my explanation is getting confusing, let me explain the use case I have in mind.

I would like to run through a folder containing subfolders with edl screens and convert them all in place, so at the end, the bob screens will be in the same locations as their edl counterparts. Example: main_folder/secondary_folder/secondary_screen.edl and secondary_screen.bob

The problem I have right now is that when launching main_screen.bob, and it references secondary_screen.bob (with no relative path information, because the original main_screen.edl depended on the EDMDATAFILES environment variables to tell it the search path for other screens), the process fails. main_screen.bob cannot find secondary_screen.bob

In short, is there a way, like with MEDM or EMD, to specify search paths for Phoebus to find .bob files, when no relative or absolute path information is given?

Thanks in advance for taking the time to clarifying this.
Max.

**Comment by kasemir** [comment]
Yes, EDM and MEDM support a search path. That is somewhat popular for a while, until you run `edm -x status.edl` and you're not sure which one it opens, but it's certainly not the one you had in mind. So then you try every `status.edl` on the search path to identify which one you get and how you can tweak the situation to get what you want.
The other issue arises when you try to make your display files available via NFS or http://... as a file server. Yes, EDM can then still use a search path with entries like `http://files.site.org/controls/main`, `http://files.site.org/controls/rf`, ... but while testing every listed entry may work on a local file system, it's slower via NFS and quite unfortunately slow via http.
While we still use EDM here at that SNS, the setup has been converted to absolute/relative paths, or if you will a search path with only one "root" entry.

The world wide web, which is overall more popular than EDM and MEDM, uses absolute or relative paths. Same with the file system. The display manager thus also uses  absolute or relative paths, no search path, so we know right away which file we should open.

When you convert EDM files, there are auto-converter options that try to deal with the search path. So there's a search path to find the file, and then you can strip a section off and place the remaining name in the destination folder.

**Reply by BoredDadDev** [reply]
Hi Kay, thanks for the quick reply. It makes sense, and going forward we can certainly make sure to use only absolute or relative paths. For now, I'll give the auto-convert with the right options a try, and see if it creates the converted .bob files in the relative paths that correspond to the original .edl path. That would mean there is not really a problem, Thanks again.

**Comment by kasemir** [comment, marked as answer]
https://github.com/ControlSystemStudio/phoebus/wiki/Display-Builder-Compatibility#paths
